### PR TITLE
workerpool: limit the rate to output deadlock warning

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve? 

Close https://github.com/pingcap/ticdc/issues/2706

### What is changed and how it works?

Ratelimit deadlock log.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

Related changes

 - Need to cherry-pick to the release branch

### Release note 

```release-note
Reduce log "synchronize is taking too long, report a bug" in some cases.
```

### CODE REVIEW

**Review by amyangfei — COMMENTED** [review]
Does this PR need a test?

File: pkg/workerpool/pool_impl.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ package workerpool
 
 import (
 	"context"
+	"golang.org/x/time/rate"
```
Comment (liuzix) [inline pkg/workerpool/pool_impl.go:18]: Should this be moved down to the second group of imports?

**Review by liuzix — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

**Pull request merged**